data/mods/Zombie_Nightvision/main.lua

```diff
@@ -0,0 +1,8 @@
+monster_types = game.get_monster_types()
+
+-- halve the speed of all monsters whose species is "ZOMBIE"
+for _, monster_type in ipairs(monster_types) do
+    if game.monster_type(monster_type):in_species("ZOMBIE") then
+        game.monster_type(monster_type).vision_min = 60 - game.monster_type(monster_type).vision_dec
```

Comment:
For this to work, you need to export those properties from `mtype` to Lua. As it's only a public class member, you need to add something like this (and similar for `vision_min`) to the mtype part of "lua/class_definitions.lua" and than compile the game with `make LUA=1` and test the changes:
```
            vision_dec = {
                type = "int",
                writable = true
            },
```